packages/mask/src/components/shared/SelectRecipients/SelectRecipients.tsx

```diff
@@ -1,62 +1,72 @@
-import { Box, Chip } from '@mui/material'
-import { makeStyles } from '@masknet/theme'
-import AddIcon from '@mui/icons-material/Add'
 import { useEffect, useState } from 'react'
-import { difference } from 'lodash-unified'
-import { useI18N } from '../../../utils'
-import type { ProfileInformation as Profile } from '@masknet/shared-base'
+import { cloneDeep } from 'lodash-unified'
+import { ProfileInformation as Profile, EMPTY_LIST, NextIDPlatform } from '@masknet/shared-base'
 import type { LazyRecipients } from '../../CompositionDialog/CompositionUI'
 import { SelectRecipientsDialogUI } from './SelectRecipientsDialog'
 import { useCurrentIdentity } from '../../DataSource/useActivatedUI'
-import { EMPTY_LIST } from '@masknet/shared-base'
-
-const useStyles = makeStyles()({
-    root: {
-        display: 'inline-flex',
-        flexWrap: 'wrap',
-    },
-})
+import { useNextIDBoundByPlatform } from '../../DataSource/useNextID'
+import { useTwitterIdByWalletSearch } from './useTwitterIdByWalletSearch'
+import { isValidAddress } from '@masknet/web3-shared-evm'
 
 export interface SelectRecipientsUIProps {
     items: LazyRecipients
     selected: Profile[]
     disabled?: boolean
     hideSelectAll?: boolean
     hideSelectNone?: boolean
+    open: boolean
+    onClose(): void
     onSetSelected(selected: Profile[]): void
 }
 
 export function SelectRecipientsUI(props: SelectRecipientsUIProps) {
-    const { t } = useI18N()
-    const { classes } = useStyles()
-    const { items, selected, onSetSelected } = props
+    const { items, selected, onSetSelected, open, onClose } = props
+    const [valueToSearch, setValueToSearch] = useState('')
     const currentIdentity = useCurrentIdentity()
+
+    const resolveNextIDPlatformType = () => {
+        return isValidAddress(valueToSearch)
```

Comment:
plz rewrite the nested ternary operation to if-else

Reply:
done